app/Http/Controllers/Api/ActivityController.php

```diff
@@ -27,11 +27,22 @@ public function index(Request $request)
     {
         // Create an empty Signup query and eager-load posts, which we
         // can either filter or paginate to retrieve all signup records.
-        $query = $this->newQuery(Signup::class)->with('posts');
+        if ($request->query('orderBy') === 'desc') {
+            $query = $this->newQuery(Signup::class)->with('posts')->orderBy('created_at', 'desc');
+        } else {
+            $query = $this->newQuery(Signup::class)->with('posts');
+        }
 
         $filters = $request->query('filter');
+
         $query = $this->filter($query, $filters, Signup::$indexes);
 
+        if (($request->query('limit') === 'all')) {
+            $signups = $query->get();
+
+            return $this->collection($signups);
```

Comment:
Should this not be paginated?

Reply:
I don't think we want it to be paginated since we want to show all the signups for a user! The issue right now is that it only returns 20 since that is the default on pagination and some users have more than 20 signups.

Reply:
gotcha